src/linkml_map/inference/schema_mapper.py

```diff
@@ -45,40 +46,40 @@ class SchemaMapper:
 
     transformer: Transformer = None
 
-    source_to_target_class_mappings: Dict[str, List[str]] = field(
+    source_to_target_class_mappings: dict[str, list[str]] = field(
         default_factory=lambda: defaultdict(list)
     )
 
-    slot_info: Dict[Tuple[str, str], Any] = field(default_factory=lambda: {})
+    slot_info: dict[tuple[str, str], Any] = field(default_factory=dict)
 
     def _copy_dict(
         self,
         copy_directive: CopyDirective,
         src_elements,
         tgt_elements,
-    ):
+    ) -> None:
         if copy_directive.copy_all:
-            for element in src_elements.keys():
+            for element in src_elements:
```

Comment:
use `for x in dict` instead of `keys()`: SIM118: https://docs.astral.sh/ruff/rules/in-dict-keys/